aws/data_source_aws_ecr_repository.go

```diff
@@ -0,0 +1,65 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsEcrRepository() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEcrRepositoryRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
```

Comment:
These redundant composite literals can be safely removed (doable since Go 1.7), so this could be:

```
"name": {
```

Could you reflect the same modification everywhere where needed? Thanks!